src/models/converters/tests/transfers.rs

```diff
@@ -165,3 +165,27 @@ fn transfer_dto_to_transaction_details() {
 
     assert_eq!(expected, actual)
 }
+
+#[test]
+fn transfer_erc721_with_null_id() {
+    let mut mock_info_provider = MockInfoProvider::new();
+    mock_info_provider.expect_safe_info().times(0);
+    mock_info_provider.expect_token_info().times(0);
+    let tx = serde_json::from_str::<Erc20TransferDto>(
+        crate::json::ERC_20_TRANSFER_WITH_ERC721_TOKEN_INFO,
+    )
+    .unwrap();
+
+    let actual = tx.to_transfer_info(&mut mock_info_provider);
+
+    let expected = TransferInfo::Erc20(Erc20Transfer {
```

Comment:
This is not the result I would expect.

Either we mark it as what it is, an ERC721 transfer, as we are transfering a erc721 token (the id is the value returned by the backend).
Or we just ignore this for now and don't convert it.

But this way we display it wrongly and in the end it says you transfered a large amount of this token.

Reply:
The `tokenId` is also `null` in this particular case. The deserialiser is already rendering this as an `Erc20Transfer`. the `type` field coming from the backend is of `"Erc20Transfer"` type so it really complicates things for us.

A solution I see is to refactor these deserialisers to be more generic and use `TokenInfo` instead?

Reply:
So `TokenInfo` defines the type of transfer, rather than the transfer itself. Although big changes here are definitely breaking changes.

Reply:
Scratch what I said, if we don't change anything in the `models::service` module there is no breaking change to be made.

Reply:
>  type so it really complicates things for us.

Then handle the throw, so that it doesn't crash the view rendering and skip this value when converting the transfer array. Proper fix can be done later (actually @Uxio0 will do that in the core services).

Reply:
I will deploy the fix tomorrow morning with some others bugfixes